src/CitizenProject/CitizenProjectUpdateCommandHandler.php

```diff
@@ -29,6 +29,27 @@ public function handle(CitizenProjectCommand $command)
             throw new \RuntimeException('A CitizenProject instance is required.');
         }
 
+        $citizenProjectSkillIdsToSubmit = [];
+        $citizenProjectSkillIdsAlreadyAssociate = [];
+
+        foreach ($command->getSkills() as $skill) {
+            $citizenProjectSkillIdsToSubmit[] = $skill->getId();
+        }
+
+        foreach ($citizenProject->getSkills() as $skill) {
+            $citizenProjectSkillIdsAlreadyAssociate[] = $skill->getId();
+        }
+
+        $citizenProjectSkillIdsToRemove = array_diff($citizenProjectSkillIdsAlreadyAssociate, $citizenProjectSkillIdsToSubmit);
+
+        foreach ($citizenProjectSkillIdsToRemove as $id) {
```

Comment:
It looks cumbersome. Why not just replacing the whole skills collection with a new one ? Any idea @HeahDude ?

Reply:
No I never really understood this part, I guess this is some business constraint I'm missing. I definitely on "cumbersome". I don't get why the CollectionType does not do this job.

Reply:
Thank you for the clarification !